PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/utils/tlp.cpp

```diff
@@ -1,15 +1,327 @@
 #include "tlp.h"
+
 int TLP::getTotalLength() {
-	int OHCLength = header->OHCVector.size()*4;
+	int OHCLength = header->OHCVector.size() * 4;
 	int header_length = header->nonBase->headerSizeInBytes;
 	int payload_length = header->lengthInDoubleWord * 4;
 	return (OHCLength + header_length + payload_length);
 }
 
 boost::dynamic_bitset<> TLP::getBitRep() {
-    boost::dynamic_bitset<> result(getTotalLength()*8);
-    result |=  (dataPayload) ;
-    result |= ((boost::dynamic_bitset<>((header->getBitRep())) << getTotalLength() ));
-    
-    return result;
+	boost::dynamic_bitset<> result(getTotalLength() * 8);
+	result |= (dataPayload);
+	result |= ((boost::dynamic_bitset<>((header->getBitRep())) << getTotalLength()));
+
+	return result;
+}
+
+/**
+ * @brief Creates a MemRead32 TLP with the given parameters.
+ *
+ * @param requesterId The requester ID of the TLP.
+ * @param tag The tag field of the TLP header.
+ * @param address The 32-bit address to read from.
+ * @param firstDWBE An array of four integers indicating which bytes are enabled in the first data word (DW).
+ * @param lastDWBE An array of four integers indicating which bytes are enabled in the last DW.
+ * @return A TLP object representing the MemRead32 TLP.
+ */
+TLP TLP::createMemRead32Tlp(int requesterId, int tag, int address, std::bitset<4>  firstDWBE, std::bitset<4> lastDWBE) {
+	TLP memRead32Tlp;
+	memRead32Tlp.header->OHCVector.push_back(new OHCA1(firstDWBE, lastDWBE));
+	memRead32Tlp.header->TLPtype = TLPType::MemRead32;
+	memRead32Tlp.header->lengthInDoubleWord = 0;
```

Comment:
I'm not sure about this line. Why are you setting this to 0?

Reply:
because it has no datapayload,  it only has a header